lib/llm/src/lora/source.rs

```diff
@@ -0,0 +1,313 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use futures::StreamExt;
+use object_store::{aws::AmazonS3Builder, path::Path as ObjectPath, ObjectStore};
+use std::{path::{Path, PathBuf}, sync::Arc};
+use url::Url;
+
+/// Minimal trait for LoRA sources
+/// Users can implement this in Rust for custom sources
+#[async_trait]
+pub trait LoRASourceTrait: Send + Sync {
```

Comment:
Would just `LoRASource` work? I prefer not to use the word `Trait` in a trait. We would have `LocalLoRASourceStruct` for example.